cmd/api/http/http_test.go

```diff
@@ -18,6 +19,8 @@ import (
 	"go.uber.org/mock/gomock"
 )
 
+var ctx context.Context = context.TODO() //Should we have a specific context to run the tests?
```

Comment:
Its OK to use context background in tests.

Reply:
> You can have two types of Context error => https://pkg.go.dev/context#pkg-constants.
When context is cancelled the request was Aborted.
When context deadline was exceeded its a Timeout.
context.Cancelled
context.DeadlineExceeded

Yes, I saw this. Thas because I created just one ErrResponse to both cases. `r.Context().Err()` will return the especific error (cancelled or deadline) occured and it will be showed at http response, added to `ErrResponseFromContext`. It simplifies the code, I see.